internal/core/command/types.go

```diff
@@ -29,14 +30,19 @@ type serviceCommand struct {
 	contract.Device
 	internal.HttpCaller
 	*http.Request
+	loggingClient logger.LoggingClient
 }
 
 // Execute sends the command to the command service
 func (sc serviceCommand) Execute() (string, int, error) {
-	LoggingClient.Debug("Issuing" + sc.Request.Method + " command to: " + sc.Request.URL.String())
+	if sc.loggingClient != nil {
```

Comment:
Commented here; applies elsewhere.

You shouldn't ever need to check for `!= nil`; serviceCommand should always have a properly initialized loggingClient field when Execute() is called.